contracts/vaults/VaultPauser/RibbonVaultPauser.sol

```diff
@@ -0,0 +1,249 @@
+     * @param _account user's address
+     * @param _amount the amount of shares
+     */
+    function pausePosition(address _account, uint256 _amount)
+        external
+        override
+    {
+        address currentVaultAddress = msg.sender;
+        IRibbonThetaVault currentVault = IRibbonThetaVault(currentVaultAddress);
+
+        // check if vault is registered
+        require(
+            registeredVaults[currentVaultAddress],
+            "Vault is not registered"
+        );
+
+        PauseReceipt storage pausedPosition =
+            pausedPositions[currentVaultAddress][_account];
+
+        // check if position is paused
+        require(
+            pausedPosition.shares == 0 && pausedPosition.round == 0,
+            "Position is paused"
+        );
+
+        uint16 round = currentVault.vaultState().round;
+
+        require(_amount < type(uint128).max, "_amount overflow");
+
+        pausedPositions[currentVaultAddress][_account] = PauseReceipt({
+            round: round,
+            shares: uint128(_amount)
+        });
+
+        emit Pause(_account, currentVaultAddress, _amount, round);
+
+        // transfer from user to pauser
+        IERC20(currentVaultAddress).safeTransferFrom(
+            _account,
+            address(this),
+            _amount
+        );
+
+        currentVault.initiateWithdraw(_amount);
+    }
+
+    /**
+     * @notice resume user's position into vault by making a deposit
+     * @param _vaultAddress vault's address
+     */
+    function resumePosition(address _vaultAddress) external override {
+        IRibbonThetaVault currentVault = IRibbonThetaVault(_vaultAddress);
+
+        // check if vault is registered
+        require(registeredVaults[_vaultAddress], "Vault is not registered");
+
+        address currentUser = address(msg.sender);
+
+        // get params and round
+        Vault.VaultParams memory currentParams = currentVault.vaultParams();
+        uint256 round = currentVault.vaultState().round;
+
+        PauseReceipt storage pauseReceipt =
+            pausedPositions[_vaultAddress][currentUser];
+        uint256 pauseReceiptRound = pauseReceipt.round;
+
+        // check if roun is closed before resuming position
+        require(pauseReceiptRound < round, "Round not closed yet");
+        uint256 totalWithdrawAmount =
+            ShareMath.sharesToAsset(
+                pauseReceipt.shares,
+                currentVault.roundPricePerShare(pauseReceiptRound),
+                currentParams.decimals
+            );
+
+        // delete position once transfer (revert to zero)
+        delete pausedPositions[_vaultAddress][currentUser];
+
+        // stETH transfers suffer from an off-by-1 error
+        // since we received STETH , we shall deposit using STETH instead of ETH
+        if (_vaultAddress == STETH_VAULT) {
+            totalWithdrawAmount = totalWithdrawAmount - 3;
+
+            emit Resume(currentUser, _vaultAddress, totalWithdrawAmount - 1);
```

Comment:
Whats the reasoning for emitting `totalWithdrawAmount - 1` rather than just `totalWithdrawAmount`

Reply:
good question, not sure, @StephKua if you don't mind chiming in

Reply:
It was because we are going to make another transfer after that, that transfer will ended up sending totalWithdrawAmount - 1, so to align with the true transfer amount, i manually - 1 when emitting the event